src/main/java/uk/gov/hmcts/ccd/domain/model/definition/SearchResultDefinition.java

```diff
@@ -28,25 +30,32 @@ public void setFields(SearchResultField[] fields) {
             .collect(Collectors.toList());
     }
 
-    public Map<String, String> getFieldsUserRoles() {
-        Map<String, String> fields = new HashMap<>();
-        for (SearchResultField srf : this.fields) {
-            fields.put(srf.getCaseFieldId(), srf.getRole());
+    public Map<String, List<String>> getFieldsUserRoles() {
+        Map<String, List<String>> fieldsUserRoles = new HashMap<>();
+        for (SearchResultField srf : fields) {
+            if (fieldsUserRoles.containsKey(srf.getCaseFieldId())) {
+                fieldsUserRoles.get(srf.getCaseFieldId()).add(srf.getRole());
+            } else {
+                fieldsUserRoles.put(srf.getCaseFieldId(), new ArrayList<>(Collections.singletonList(srf.getRole())));
+            }
         }
-        return fields;
+        return fieldsUserRoles;
     }
 
     public boolean fieldExists(String caseFieldId) {
-        Map<String, String> fields = getFieldsUserRoles();
+        Map<String, List<String>> fields = getFieldsUserRoles();
         return fields.containsKey(caseFieldId);
     }
 
     public boolean fieldHasRole(String caseFieldId, Set<String> roles) {
-        Map<String, String> fields = getFieldsUserRoles();
-        String role = fields.get(caseFieldId);
-        if (role != null && !roles.contains(role)) {
-            return false;
+        Map<String, List<String>> fields = getFieldsUserRoles();
+        List<String> userRoles = fields.get(caseFieldId);
+        if (userRoles != null) {
+            long count = userRoles.stream()
+                .filter(userRole -> userRole == null || roles.contains(userRole))
+                .count();
+            return count != 0;
         }
-        return true;
+        return false;
```

Comment:
there is change in behaviour here when the field has no user roles. In the previous version of the code `fieldHasRole` would return `true`, now it returns `false`. Is that what correct?

Reply:
Not quite. Before we would return false if there was a non null role that was not contained in the roles set - we would return true for null roles and those that were contained in the roles set.

With my change, we return false if there is no mapping to the caseFieldId (therefore no roles not even a null one) or if there is a mapping to the caseFieldId but none of the roles in that mapping exist in the roles set or none of them are null (the count will therefore be 0 and the condition will be false and return false).